2019Labs/CustomSecurityContent/documentation/lab0_setup-devconf.adoc

```diff
@@ -111,4 +111,83 @@ The extension displays previews of the web page next to occurrences of the expre
+== 0.4 Setting up the Lab Environment using Fedora Virtual Machine
+
+NOTE: If you use the provided lab environment you can skip this entire section.
+
+This section contains instructions on how to setup a Fedora environment for the lab.
+
+=== 0.4.1 Creating a Fedora Virtual Machine
+
+. Download latest available version of Fedora ISO image here: https://getfedora.org/en/workstation/download/
+
+. Install it using your favorite Virtual Machine software (e.g. `KVM`, `VirtualBox`, `VMWare`, etc...)
+
+. After you complete the installation and everything is set up, start the `SSH` server on the Virtual Machine. Open a terminal on the Virtual Machine and type:
++
+----
+[... ~]$ sudo service sshd start
+----
++
+. Go to `Settings->Network` and write down the IPv4 address of this Virtual Machine so we can run Ansible^(R)^ playbooks to setup the lab environment and also to access the terminal through `SSH`. The following image illustrates how to find the IPv4 address:
+
+image:0-06-ipaddress.png[600,600]
+
+=== 0.4.2 Setting up the Lab environment
+
+. Now on your host machine, clone this repository to access the lab setup playbooks (`git` package needs to be installed):
++
+----
+[... ~]$ git clone https://github.com/RedHatDemos/SecurityDemos.git
+----
++
+
+. Switch to the right directory:
++
+----
+[... ~]$ cd SecurityDemos/2019Labs/CustomSecurityContent/setup
+----
++
+
+. Now it is time to run the Ansible^(R)^ playbooks. But first make sure that you have Ansible^(R)^ installed on your machine, follow the instructions here first if you don't have: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html
+
++
+----
+[... setup]$ ansible-playbook -i <VM_IPv4_ADDRESS>, system_setup.yml labs_setup.yml -k -K -u <VM_USER>
+----
++
+
+NOTE: Replace `<VM_IPv4_ADDRESS>` with the Virtual Machine's IPv4 address and `<VM_USER>` with the user you created in your Virtual Machine. Pay attention at the required `comma` after the IPv4 address parameter. You will be prompted to input the password of `<VM_USER>` twice, the second time you can press `Enter` so it uses the first password as input.
+
+. After the `ansible-playbook` command finishes, you are now able to start the lab. First login to the user `lab-user` (the setup playbooks have created it) on you Virtual Machine using the password *r3dh4t1!*. And for convenience you can run all the commands through a `SSH` terminal, run the following command:
+
++
+----
+[... setup]$ ssh lab-user@<VM_IPv4_ADDRESS>
+----
++
+
+Use password:
+
+----
+...password: r3dh4t1!
+----
+
+=== Important thing to keep in mind (Fedora only)
+
+. This lab was primarily created to work on Red Hat^(R)^ Enterprise Linux^(R)^ machines, but it is totally compatible with Fedora. So whenever you see `rhel8` or `rhel7` in commands that need to be run you switch to `fedora`. For example:
++
+----
+[... ~]$ ./build_product rhel8
+----
++
+
+But instead you execute:
++
+----
+[... ~]$ ./build_product fedora
+----
++
+
```

Comment:
I would add that this applies to more than that - we also look elsewhere for build artifacts and for profile files (can't think of anything else)

Reply:
I slightly changed the text and added more examples.